File: googleProblems/House Robber.java
class Solution {
    public int rob(int[] nums) {
        int total1 = 0;
        int total2 = 0;
        int j = nums.length;
        for (int i = 0; i < nums.length; i += 2) {
            total1 += nums[i];
            total2 += nums[j];
            j -= 2;
        }

        return Math.max(total1, total2);
    }
}

File: easy/House robber.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Previous max holds the most recent sum of robberies
        # Previous max holds the sum two houses before
        prev_max = prev_prev_max = tmp = 0
        for num in nums:
            prev_prev_max = tmp
            tmp = prev_max
            # Max is either the most recent sum, or the sum of robberies
            # two houses ago.
            prev_max = max(prev_max, prev_prev_max + num)
        return prev_max